app/models/course/condition.rb

```diff
@@ -5,4 +5,9 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  ALL_CONDITIONS = [
+    Course::Condition::Achievement.class_name,
```

Comment:
Final change, this should be `Course::Condition::Achievement.class`, remember the `Class.class_name` is an extension from Yardoc and won't be in production builds.

Also see the partial where you list all the condition types, I have another change there